### PR TITLE
Mail: Filter recipient variables

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Related to #2949

#### What's this PR do?
Convert users inserted variables like `[PreferredName]` to mailgun recipient variables `%recipient.preferred_name%`

#### How should this be manually tested?
Send an email to your user (with your email address). Use [PreferredName] and [Email] in the header or body of the email. Make sure your variable `MAILGUN_RECIPIENT_OVERRIDE` is not set.

### CODE REVIEW

File: mail/utils.py [hunk 1]
```diff
@@ -85,3 +85,21 @@ def generate_mailgun_response_json(response):
...
+
+def filter_recipient_variables(text):
+    """
+    Filter out recipient variables, like [PreferredName], and substitute it with %recipient.preferred_name%
+    Args:
+        text (string): subject or body of the email
+    Returns:
+        string: with replaced correct recipient variables
+    """
+    substitute_values = {
```
Comment (rhysyngsun) [inline mail/utils.py]: This could be a module-level constant.

File: mail/utils.py [hunk 2]
```diff
@@ -85,3 +85,21 @@ def generate_mailgun_response_json(response):
             "message": response.reason
         }
     return response_json
+
+
+def filter_recipient_variables(text):
```
Comment (rhysyngsun) [inline mail/utils.py]: Could you write a test or two for this, I think it's technically covered by other tests, but I don't think the expected behavior is being asserted.

**Comment by rhysyngsun** [comment]
Functionality is good.

File: mail/utils.py (hunk 2 above)
Comment (annagav) [inline mail/utils.py]: I added a test

**Pull request merged**